### ISSUE TITLE
Bump vega typings to 1.0

### ISSUE THREAD

**Issue by domoritz** [vega/vega#3329]
We should bump the vega tyings to 1.0 in the next release. This will signal to users that they can start to rely on the types and that we will adhere to semantic versioning.

**Comment by danmarshall** [comment]
As of 0.22.0 they no longer work with TypeScript < 4

**Comment by thw0rted** [comment]
Please update the package file for `vega-typings` to specify `typescript` as a peer dependency, with a minimum version corresponding to the features that are used by the emitted typings.

I just spent a bunch of time troubleshooting why our build broke.  I inherited a project that's stuck on an older Typescript version (3.9), and when somebody updated the version of `vega-typings`, it added type definitions that use template literal string types -- these were actually added in TS 4.1, so even 4.0 wouldn't be good enough.

(It would be great if you could consider automating tests of your library against a variety of older TS versions, too!)

**Comment by domoritz** [comment]
Good idea. We actually test against multiple versions of ta already and specify the minimum version in packages/vega-typings/index.d.ts.
